components/service/sync-autofill/src/test/java/mozilla/components/service/sync/autofill/AutofillCreditCardsAddressesStorageTest.kt

```diff
@@ -66,82 +77,121 @@ class AutofillCreditCardsAddressesStorageTest {
         )
-        val creditCardFields2 = UpdatableCreditCardFields(
+        val plaintextNumber2 = CreditCardNumber.Plaintext("4111111111111111")
+        val creditCardFields2 = NewCreditCardFields(
             billingName = "Banana Apple",
-            cardNumber = "4111111111111112",
+            plaintextCardNumber = plaintextNumber2,
+            cardNumberLast4 = "1111",
             expiryMonth = 1,
             expiryYear = 2030,
-            cardType = "discover"
+            cardType = "visa"
         )
-        val creditCardFields3 = UpdatableCreditCardFields(
+        val plaintextNumber3 = CreditCardNumber.Plaintext("340000000000009")
+        val creditCardFields3 = NewCreditCardFields(
             billingName = "Pineapple Orange",
-            cardNumber = "4111111111111113",
+            plaintextCardNumber = plaintextNumber3,
+            cardNumberLast4 = "0009",
             expiryMonth = 2,
             expiryYear = 2028,
-            cardType = "visa"
+            cardType = "amex"
         )
         val creditCard1 = storage.addCreditCard(creditCardFields1)
         val creditCard2 = storage.addCreditCard(creditCardFields2)
         val creditCard3 = storage.addCreditCard(creditCardFields3)
 
         creditCards = storage.getAllCreditCards()
 
+        val key = storage.crypto.key()
+
         assertEquals(3, creditCards.size)
         assertEquals(creditCard1, creditCards[0])
+        assertEquals(plaintextNumber1, storage.crypto.decrypt(key, creditCards[0].encryptedCardNumber))
         assertEquals(creditCard2, creditCards[1])
+        assertEquals(plaintextNumber2, storage.crypto.decrypt(key, creditCards[1].encryptedCardNumber))
         assertEquals(creditCard3, creditCards[2])
+        assertEquals(plaintextNumber3, storage.crypto.decrypt(key, creditCards[2].encryptedCardNumber))
     }
 
     @Test
-    fun `updateCreditCard`() = runBlocking {
-        val creditCardFields = UpdatableCreditCardFields(
+    fun `update credit card`() = runBlocking {
+        val creditCardFields = NewCreditCardFields(
             billingName = "Jon Doe",
-            cardNumber = "4111111111111111",
+            plaintextCardNumber = CreditCardNumber.Plaintext("4111111111111111"),
+            cardNumberLast4 = "1111",
             expiryMonth = 12,
             expiryYear = 2028,
-            cardType = "amex"
+            cardType = "visa"
         )
 
         var creditCard = storage.addCreditCard(creditCardFields)
 
-        val newCreditCardFields = UpdatableCreditCardFields(
+        // Change every field
+        var newCreditCardFields = UpdatableCreditCardFields(
             billingName = "Jane Fields",
-            cardNumber = "4111111111111112",
+            cardNumber = CreditCardNumber.Plaintext("30000000000004"),
+            cardNumberLast4 = "0004",
             expiryMonth = 12,
             expiryYear = 2038,
-            cardType = "visa"
+            cardType = "dinersclub"
```

Comment:
Desktop uses "diners". https://searchfox.org/mozilla-central/source/toolkit/modules/CreditCard.jsm#15

I am planning to keep the cardTypes consistent with desktop.

```suggestion
            cardType = "diners"
```

Reply:
Ok!